sdk/src/client/api/block_builder/input_selection/requirement/mana.rs

```diff
@@ -7,7 +7,16 @@ use crate::client::secret::types::InputSigningData;
 impl InputSelection {
     pub(crate) fn fulfill_mana_requirement(&mut self, allotments: u64) -> Result<Vec<InputSigningData>, Error> {
         let required_mana = self.outputs.iter().map(|o| o.mana()).sum::<u64>() + allotments;
-        let mut selected_mana = self.selected_inputs.iter().map(|o| o.output.mana()).sum::<u64>();
+        let mut selected_mana = 0;
+
+        for input in &self.selected_inputs {
+            selected_mana += input.output.available_mana(
```

Comment:
any reason you think we should do safe adds?

Reply:
There is an open issue to evaluate if we should safe add/sub in ISA so I'd defer that decision to it. I'm not fully sure it's needed as it's already done in semantic check so we can keep ISA lighter.

Reply:
okay!